app/app.go

```diff
@@ -1059,12 +1120,65 @@ func New(
 		app.GetTxConfig().TxEncoder(),
 		mempool,
 	)
-	app.SetPrepareProposal(handler.PrepareProposalHandler())
 
-	// we use a no-op ProcessProposal, this way, we accept all proposals in avoidance
-	// of liveness failures due to Prepare / Process inconsistency. In other words,
-	// this ProcessProposal always returns ACCEPT.
-	app.SetProcessProposal(baseapp.NoOpProcessProposal())
+	// Read general config from app-opts, and construct oracle service.
+	cfg, err := oracleconfig.ReadConfigFromAppOpts(appOpts)
+	if err != nil {
+		panic(err)
+	}
+
+	// If app level instrumentation is enabled, then wrap the oracle service with a metrics client
+	// to get metrics on the oracle service (for ABCI++). This will allow the instrumentation to track
+	// latency in VerifyVoteExtension requests and more.
+	oracleMetrics, err := servicemetrics.NewMetricsFromConfig(cfg, app.ChainID())
+	if err != nil {
+		panic(err)
+	}
+
+	// Create the oracle service.
+	app.oracleClient, err = oracleclient.NewClientFromConfig(
+		cfg,
+		app.Logger().With("client", "oracle"),
+		oracleMetrics,
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	// Connect to the oracle service (default timeout of 5 seconds).
+	go func() {
+		// TODO: change background to something cancellable across the app?
+		if err := app.oracleClient.Start(context.Background()); err != nil {
+			app.Logger().Error("failed to start oracle client", "err", err)
+			panic(err)
+		}
+
+		app.Logger().Info("started oracle client", "address", cfg.OracleAddress)
+	}()
```

Comment:
about point number 3 - app.oracleClient.Start is calll syncronous? and if it fails to run, it will return err != nil and crush. Or I'm missing something

Reply:
it is sync according to the implementation: https://github.com/skip-mev/slinky/blob/4c45a1de0482d2d1f52e2a174dcbc86db942dec7/service/clients/oracle/client.go#L108. it is sync, but it [doesn't wait](https://github.com/skip-mev/slinky/blob/4c45a1de0482d2d1f52e2a174dcbc86db942dec7/service/clients/oracle/client.go#L126) until the connection is established unless we configure it with an option to perform [BlockingDial](https://github.com/skip-mev/slinky/blob/4c45a1de0482d2d1f52e2a174dcbc86db942dec7/service/clients/oracle/options.go#L9. I think it makes sense to create a context with some timeout and pass it to the `oracleClient.Start` method and use the BlockingDial

Reply:
please doublecheck my suggestions and make a kind of (manual?) test covering the scenarios of unavailable oracle